sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonUtils.scala

```diff
@@ -55,4 +60,26 @@ object JacksonUtils {
 
     schema.foreach(field => verifyType(field.name, field.dataType))
   }
+
+  private def validateStringLiteral(exp: Expression): String = exp match {
+    case Literal(s, StringType) => s.toString
+    case e => throw new AnalysisException("Must be a string literal, but: " + e)
+  }
+
+  def validateSchemaLiteral(exp: Expression): StructType =
+    DataType.fromJson(validateStringLiteral(exp)).asInstanceOf[StructType]
```

Comment:
I just wrote this way along with here https://github.com/apache/spark/blob/master/sql/core/src/main/scala/org/apache/spark/sql/functions.scala#L3010. Both is okay to me though, if we modify the code in a way you suggested, we need to modify `from_json` code, too?